src/Stats.h

```diff
@@ -0,0 +1,27 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+namespace reactivesocket {
+// forward decl for NoopStats
+class NoopStats;
+
+// Virtual interface for a stats receiver
+class Stats {
+ public:
+  static NoopStats noop;
```

Comment:
We might regret this. Better to have a static in a method and return reference to it (so the static gets initialised lazily on the first invocation). Or a proper singleton.